src/cluster.ts

```diff
@@ -690,35 +697,42 @@ Please use the format 'my-cluster' or '${instance.name}/clusters/my-cluster'.`);
     gaxOptionsOrCallback?: CallOptions | SetClusterMetadataCallback,
     cb?: SetClusterMetadataCallback
   ): void | Promise<SetClusterMetadataResponse> {
+    ClusterUtils.validateClusterMetadata(metadata);
     const callback =
       typeof gaxOptionsOrCallback === 'function' ? gaxOptionsOrCallback : cb!;
     const gaxOptions =
       typeof gaxOptionsOrCallback === 'object'
         ? gaxOptionsOrCallback
         : ({} as CallOptions);
-
-    const reqOpts: ICluster = Object.assign(
-      {},
-      {
-        name: this.name,
-        serveNodes: metadata.nodes,
-      },
-      metadata
-    );
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    delete (reqOpts as any).nodes;
-
-    this.bigtable.request<Operation>(
-      {
-        client: 'BigtableInstanceAdminClient',
-        method: 'updateCluster',
-        reqOpts,
-        gaxOpts: gaxOptions,
-      },
-      (err, resp) => {
-        callback(err, resp);
-      }
-    );
+    const setMetadataWithLocation = (location: string, name: string) => {
```

Comment:
I would remove this logic if we think it's currently not needed based on your testing.